frontend/packages/ceph-storage-plugin/src/components/create-storage-system/footer.tsx

```diff
@@ -192,34 +156,26 @@ const handleReviewAndCreateNext = async (
     } else if (type === BackingStorageType.EXTERNAL) {
       const { createPayload, model, displayName } = getExternalStorage(externalStorage) || {};
 
-      const firstExternalSystemName = getExternalSubSystemName(displayName);
-      const hasAnExternalSystem = externalSystem?.metadata?.name === firstExternalSystemName;
-
-      const subSystemName = isRhcs
-        ? OCS_EXTERNAL_CR_NAME
-        : hasAnExternalSystem
-        ? `${firstExternalSystemName}-${storageClass.name}`.substring(0, 230)
-        : firstExternalSystemName;
+      const externalSystemName = getExternalSubSystemName(displayName, storageClass?.name);
 
+      const subSystemName = isRhcs ? OCS_EXTERNAL_CR_NAME : externalSystemName;
       const subSystemState = isRhcs ? connectionDetails : createStorageClass;
       const subSystemKind = getStorageSystemKind(model);
-
       const subSystemPayloads = createPayload(
         subSystemName,
         subSystemState,
         model,
         storageClass.name,
       );
 
-      await createExternalSubSystem(subSystemPayloads);
       await labelOCSNamespace();
-      // Create storage cluster if one is not present except for external RHCS
+      await createStorageSystem(subSystemName, subSystemKind);
       if (!hasOCS && !isRhcs) {
         await labelNodes(nodes);
         await createStorageCluster(state);
       }
-      // Create a new storage system for non RHCS external vendor if one is not present
-      if (hasAnExternalSystem) await createStorageSystem(subSystemName, subSystemKind);
+      if (!isRhcs) await waitforCRD([model.plural, model.apiGroup].join('.'), model.plural);
```

Comment:
```suggestion
      if (!isRhcs) await waitforCRD(model), model.plural);
```
`waitForCRD` function is not a generic function and it has a specific use case hence it should be modeled to make things cleaner. A single model passed should suffice both the two parameters.

Reply:
The idea is to make pass crd name and crd plural.
We dont need all values from model.

Reply:
You can always pick it up from the object. My comment is based around the idea that the function signature will be cleaner.